router/README.md

```diff
@@ -0,0 +1,233 @@
+# React Router
+
+`@bhoos/react-kit-router` is the minimal implementation of router module for react native apps.
+
+This is the documentation for v2.0.0. Documentation for older version is [here](../docs/Router.md)
+
+## Getting started
+
+There are no dependencies for this package. Just add the package in your existing react-native project.
+```bash
+yarn add @bhoos/react-kit-router
+```
+
+
+## Usage
+
+This package exposes three major things:
+  - `Route` - for defining routes where components resides with specific urls
+  - `Outlet` - Outlet for the url, which is more like a placeholder of the components
+  - `useCurrentRoute` - This hook gives information about the routes that we have defined in `Route`
+
+
+For react-kit-router to work, you need to wrap your App with `withRouter` in order to use `Route`.
+
+### Example:
+
+```tsx
+import { useCurrentRoute, withRouter } from '@bhoos/react-kit-router';
+
+type PathParams = {
+  videoId: string
+}
+
+type QueryParams = {
+  t?: string
+}
+
+const mapRoutes = {
+  // VP is an example video player component with arbitary implementation.
+  'videoPlayer/:videoId':
+      ({ videoId }: PathParams, { t }: QueryParams) => () => <VP videoId={videoId} t={t} />,
+  '': () => App,
+}
+
+export const App = withRouter(
+    driver,
+    mapRoutes,
+    '', // default path for the router
+  )(() => {
+    const Route = useCurrentRoute();
+    return <Route />;
+  });
```

Comment:
@sammelanyogi Is there any reason why this cannot be 


```
<RouterProvider mapRoutes={mapRoutes} getInitialUrl={getInitialUrl} subscribe={subscribe}>
</RouterProvider>
```

instead of using `withRouter`.